spyder_env_manager/spyder/widgets/main_widget.py

```diff
@@ -0,0 +1,408 @@
+    )
+).read()
+CSS_PATH = osp.join(PLUGINS_PATH, "help", "utils", "static", "css")
+
+
+class SpyderEnvManagerWidgetActions:
+    # Triggers
+    SelectEnvironment = "select_environment"
+    NewEnvironment = "new_environment"
+    DeleteEnvironmentToolbar = "delete_environment"
+    InstallPackageToolbar = "install_package"
+
+    # Options menu actions
+    ImportEnvironment = "import_environment_action"
+    ExportEnvironment = "export_environment_action"
+    ToggleExcludeDependency = "exclude_dependency_action"
+
+
+class SpyderEnvManagerWidgetOptionsMenuSections:
+    Display = "excludes_section"
+    Highlight = "highlight_section"
+
+
+class SpyderEnvManagerWidgetMainToolBarSections:
+    Main = "main_section"
+
+
+# =============================================================================
+# ---- Widgets
+# =============================================================================
+
+
+class SpyderEnvManagerWidget(PluginMainWidget):
+
+    # PluginMainWidget class constants
+    ENABLE_SPINNER = True
+
+    def __init__(self, name=None, plugin=None, parent=None):
+        super().__init__(name, plugin, parent)
+
+        conda_env = get_list_conda_envs_cache()
+        pyenv_env = get_list_pyenv_envs_cache()
+        envs = {**conda_env, **pyenv_env, "No environments available": ""}
+        # envs={}
+
+        self.select_environment = QComboBox(self)
+        # self.create_combobox(
+        #    _('Recent custom interpreters'),
+        #    self.get_option('custom_interpreters_list'),
+        #    'custom_interpreter',
+        # )
+        self.select_environment.ID = SpyderEnvManagerWidgetActions.SelectEnvironment
+        # if not envs:
+        #    envs={'No environments available'}
+
+        self.select_environment.addItems(envs)
+
+        self.select_environment.setToolTip("Select an environment")
+        self.select_environment.setMinimumWidth(680)
+        self.css_path = self.get_conf("css_path", CSS_PATH, "appearance")
+
+        self.infowidget = FrameWebView(self)
+        self.table_layout = EnvironmentPackagesTable(self, text_color=ima.MAIN_FG_COLOR)
+        self.stack_layout = layout = QStackedLayout()
+        layout.addWidget(self.infowidget)
+        layout.addWidget(self.table_layout)
+        self.setLayout(self.stack_layout)
+        # Signals
+        self.select_environment.currentIndexChanged.connect(
+            lambda x: self.source_changed()
+        )
+
+    # ---- PluginMainWidget API
+    # ------------------------------------------------------------------------
+    def get_title(self):
+        return _("Spyder Env Manager")
+
+    def setup(self):
+        # ---- Options menu actions
+
+        exclude_dependency_action = self.create_action(
+            SpyderEnvManagerWidgetActions.ToggleExcludeDependency,
+            text=_("Exclude dependency packages"),
+            tip=_("Exclude dependency packages"),
+            toggled=lambda checked: self.packages_dependences(checked),
```

Comment:
For this action and the ones bellow, could you check doing a lambda call is needed instead of just passing the corresponding method? Like for example here this line could become:

```python
    toggled=self.packages_dependences,
```